### PR TITLE
[TB-3003] Implement dots indicator widget

### PR DESCRIPTION
**Pull request by ghost**
### What 🕵️ 🔍

- What does the pull request solve?
- What exactly was added or modified?
It implements the `DotsIndicator` widget, that can be used in combination with a PageView for different scenarios - for example for the onboarding

----------

### References 📝 🔗

- [TB-2998](https://xainag.atlassian.net/browse/TB-2998)
- [TB-3003](https://xainag.atlassian.net/browse/TB-3003)

----------

### CODE REVIEW

File: example/lib/screen/xayn_widgets/page/dots_indicator_page.dart [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+        onPrimary: () => linden.colors.dotsIndicatorActiveColor,
+        onSecondary: () => linden.colors.secondaryAccent,
+        onTertiary: () => linden.colors.primaryAction,
+      );
+
+  Color get getDotsBorderColor => dotsBorderColor.fold(
+        onPrimary: () => linden.colors.dotsIndicatorBorderColor,
+        onSecondary: () => linden.colors.secondaryAccent,
+        onTertiary: () => linden.colors.primaryAction,
+      );
```
Comment (pro100svitlo) [inline example/lib/screen/xayn_widgets/page/dots_indicator_page.dart:286]: 💅 those looks exactly the same, so maybe we do not need an extension?
```dart
Color getColor(_DotsColor color){
  switch(color){
    case _DotsColor.primary:
        return linden.colors.dotsIndicatorBorderColor;
      case _DotsColor.secondary:
        return linden.colors.secondaryAccent;
      case _DotsColor.tertiary:
        return linden.colors.primaryAction;
  }
}
```

this one should be more then enough 😉

File: lib/src/widget/dots_indicator/dots_indicator.dart [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      borderWidth: linden.dimen.unit0_25,
+      shape: BoxShape.circle,
+    );
+  }
+
+  Widget _buildDot(int index, DotsDecorator decorator) {
+    final color =
+        index == activePosition ? decorator.activeColor : decorator.color;
+
+    final dot = Container(
```
Comment (pro100svitlo) [inline lib/src/widget/dots_indicator/dots_indicator.dart:70]: 💅 return can be done here without `dot` variable

File: example/lib/screen/xayn_widgets/page/dots_indicator_page.dart [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+        onChanged: (value) {
+          setState(() {
+            dotsShape = value!;
+          });
+        },
+      ),
+    );
+  }
+
+  Widget _buildListTileDotsSpacing() {
```
Comment (pro100svitlo) [inline example/lib/screen/xayn_widgets/page/dots_indicator_page.dart:227]: 💅 `{}` can be omitted for instant `return`:
```dart
  Widget _buildListTileDotsSpacing() => ListTile(
```
please check other places as well 🙏

**Review by pro100svitlo — COMMENTED** [review]
Looks good to me, but left few  💅  commenst

File: example/lib/screen/xayn_widgets/page/dots_indicator_page.dart (hunk 1 above)
Comment (carminetoriello) [reply example/lib/screen/xayn_widgets/page/dots_indicator_page.dart:286]: Yeep, will do it, thanks!!

Comment (carminetoriello) [reply example/lib/screen/xayn_widgets/page/dots_indicator_page.dart:286]: Sorry, they are actually different for the primary color that might be color, activeColor and borderColor - but still I have an improvement that I'm going to try ;)

**Review by pro100svitlo — APPROVED** [review]

**Pull request merged**